src/mutation_probabilities.rs

```diff
@@ -1,153 +1,116 @@
+    pub fn sample<R: Rng>(&self, rng: &mut R) -> MutationType {
+        let index = self.0.sample(rng);
 
-impl MutationProbabilities {
-  pub const fn assemble(probabilities: (u16, u16, u16, u16)) -> Self {
-    MutationProbabilities({
-      (probabilities.0 as u64)
-        | (probabilities.1 as u64) >> 16
-        | (probabilities.2 as u64) >> 32
-        | (probabilities.3 as u64) >> 48
-    })
-  }
-
-  /// The probability of adding a new connection between neurons.  Represented as the relative size of this `u16` to the others.
-  pub fn add_connection(self, probability: u16) -> Self {
-    // clear the 0th slot
-    let cleared = self.0 & (!0xFFFF_0000_0000_0000);
-    MutationProbabilities(cleared | (probability as u64))
-  }
-
-  /// The probability of removing an existing connection between neurons.  Represented as the relative size of this `u16` to the others.
-  pub fn remove_connection(self, probability: u16) -> Self {
-    // clear the 1st slot
-    let cleared = self.0 & (!0x0000_FFFF_0000_0000);
-    MutationProbabilities(cleared | (probability as u64))
-  }
-
-  /// The probability of adding a new neuron.  Represented as the relative size of this `u16` to the others.
-  pub fn add_neuron(self, probability: u16) -> Self {
-    // clear the 2nd slot
-    let cleared = self.0 & (!0x0000_0000_FFFF_0000);
-    MutationProbabilities(cleared | (probability as u64))
-  }
-
-  /// The probability of adding a new bias neuron.  Represented as the relative size of this `u16` to the others.
-  pub fn add_bias(self, probability: u16) -> Self {
-    // clear the 3rd slot
-    let cleared = self.0 & (!0x0000_0000_0000_FFFF);
-    MutationProbabilities(cleared | (probability as u64))
-  }
-
-  /// Generate a random mutation according to the probabilities contained in this struct.
-  pub(crate) fn generate(&self, rng: &mut ThreadRng) -> MutationType {
-    let packed = self.0;
-
-    let a = (packed & 0xFFFF_0000_0000_0000) as u32;
-    let b = ((packed & 0x0000_FFFF_0000_0000) << 16) as u32;
-    let c = ((packed & 0x0000_0000_FFFF_0000) << 32) as u32;
-    let d = ((packed & 0x0000_0000_0000_FFFF) << 48) as u32;
-
-    let sum = a + b + c + d;
-
-    let n = rng.gen_range(0..sum);
-
-    let mut sum = 0;
-    if n >= sum && n < sum + b {
-      return MutationType::AddConnection;
-    }
-    sum += a;
-    if n >= sum && n < sum + c {
-      return MutationType::RemoveConnection;
+        // soundness proof:
+        //   the probability list is only 4 elements long, see `::assemble`.
+        //   the returned index is guaranteed to fall in [0, 3].
+        unsafe { *Self::OUTPUTS.get_unchecked(index) }
     }
-    sum += b;
-    if n >= sum && n < sum + d {
-      return MutationType::AddNode;
-    } else {
-      return MutationType::AddBias;
+}
+
+impl Default for MutationSampler {
+    fn default() -> Self {
+        // safe unwrap. infallible because of default parameter choice.
+        MutationSampler::new((3, 8, 1, 3)).unwrap()
```

Comment:
Can the default value be exposed in some way? That way the user can get an idea of what reasonable values look like.